### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by varqox**
Some new feature

### CODE REVIEW

File: src/static/kit/scripts.js [hunk 1]
```diff
@@ -826,6 +826,10 @@ function default_tabmenu_attacher(x) {
 	x.on('tabmenuTabHasChanged', function() { x.nextAll().remove(); });
 	x.appendTo(this);
 }
+function tabmenu_attacher_with_change_callback(tabmenu_changed_callback, x) {
+	x.in('tabmenuTabHasChanged', tabmenu_changed_callback);
```
Comment (tropuq) [inline src/static/kit/scripts.js]:
```suggestion
	x.on('tabmenuTabHasChanged', tabmenu_changed_callback);
```
I think that is better

File: src/static/kit/scripts.js [hunk 2]
```diff
@@ -1322,9 +1326,10 @@ function Logs(type, elem, auto_refresh_checkbox) {
 		var curr_height = content[0].scrollHeight;
 		content.scrollTop(curr_height - prev);
 
-		// Load more logs if scrolling up did not become possible
+		lock = false;
+
+		// Load more logs unless scrolling up became impossible
 		if (offset > 0) {
-			lock = false;
```
Comment (tropuq) [inline src/static/kit/scripts.js]: It is not ok

File: src/web_interface/submissions_api.cc [hunk 3]
```diff
@@ -432,36 +432,29 @@ void Sim::api_submissions() {
...
-							            ? NORMAL
-							            : problem_final_to_subtype());
-						} else {
-							subtype_to_show =
-							   (is_contest_initial_final ? INITIAL_FINAL
-							                             : NORMAL);
-						}
-						break;
-					}
 					case ScoreRevealingMode::SCORE_AND_FULL_STATUS: {
```
Comment (tropuq) [inline src/web_interface/submissions_api.cc]:
```suggestion
					case ScoreRevealingMode::SCORE_AND_NOT_FULL_STATUS: {
```

File: src/static/kit/scripts.js (hunk 2 above)
Comment (varqox) [reply src/static/kit/scripts.js]: It is ok!

**Pull request closed without merging**